### PR TITLE
Refactoring to remove duplicate code.

### PR DESCRIPTION
**Pull request by fmaximus**
## Description

* Removing Code Duplication by refactoring Admin Commands.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

## How Has This Been Tested?

Added new Unit Tests.

### CODE REVIEW

**Comment by rhtyd** [comment]
@fmaximus master is unfrozen now, can you fix the conflict. We can merge this after running a final regression test. Thanks.

**Comment by fmaximus** [comment]
Sorry for the delay. I've resolved the conflicts.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
Hi @DaanHoogland - can you help review and stabilize this PR? Thanks.

**Comment by fmaximus** [comment]
I've rebased. I'm still monitoring this PR, and will continue to do so until it's merged.

**Comment by rhtyd** [comment]
Thanks @fmaximus 
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by rhtyd** [comment]
Pinging for review @anuragaw @shwstppr @DaanHoogland and others.

File: api/src/main/java/org/apache/cloudstack/api/command/user/UserCmd.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// under the License.
+///
+
+package org.apache.cloudstack.api.command.user;
+
+import org.apache.cloudstack.api.ResponseObject.ResponseView;
+import org.apache.cloudstack.api.command.ResponseViewProvider;
+
+public interface UserCmd extends ResponseViewProvider {
+    default ResponseView getResponseView() {
```
Comment (anuragaw) [inline api/src/main/java/org/apache/cloudstack/api/command/user/UserCmd.java]: nit- perhaps it's just me but I think getResponseViewType can be better to read ?

**Comment by shwstppr** [comment]
@fmaximus can you please resolve conflicts

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
cc @shwstppr @borisstoyanov @anuragaw

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@blueorangutan test centos7 xenserver-71

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-65u2

**Comment by rhtyd** [comment]
Reopened it here: https://github.com/apache/cloudstack/pull/3538

**Pull request merged**